cmake/googletest/common.cmake

```diff
@@ -13,20 +13,19 @@ function(googletest_get_populate)
         add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR})
     endif()
 
-    # Disable compiler warnings and clang-tidy
-    if(DISABLE_EXTERNAL_WARNINGS)
-        target_compile_options(gtest PRIVATE -w)
-        set_target_properties(gtest PROPERTIES CXX_CLANG_TIDY "")
+endfunction()
 
-        target_compile_options(gtest_main PRIVATE -w)
-        set_target_properties(gtest_main PROPERTIES CXX_CLANG_TIDY "")
+function(googletest_disable_warnings)
```

Comment:
Why another function? I assume there is a reason, but isn't googletest_get_populate only called if it's not provided anyway?

Reply:
If gtest is provided by catkin, the get_populate won't be called. But, catkin provides the sources and not a binary. So we must disable warnings on what has been provided. This is done exactly the same in HiveMind